tsconfig.json

```diff
@@ -0,0 +1,18 @@
+{
+  "compilerOptions": {
+    "outDir": "./built",
+    "allowJs": true,
+    "target": "es6",
+    "moduleResolution": "node16",
+    "allowSyntheticDefaultImports": true,
+    "sourceMap": true,
+    "esModuleInterop": true,
+    "module": "node16", // Imporant. No other value really works for us. 'commonjs' does not play well with dynamic import(). 'esnext' pretty much require to migrate everything
```

Comment:
I was about to give up on the whole migration and then found this module setting. The two most common ones used are 'esnext' and 'commonjs'. Using the former pretty much requires you to migrate everything to `.ts` files and the project to ESM, which would have turned the PR into a massive undertaking. Using the latter does not work because we depend on ESM only packages (p-queue and p-retry) and they don't work in this setup.